[GCC: 10.2.0] build error when compiling Identity.cc

```
[  4%] Building CXX object src/CMakeFiles/ignition-physics3.dir/Identity.cc.o
In file included from /home/acxz/vcs/git/github/acxz/pkgbuilds/ignition-physics/src/ign-physics-ignition-physics3_3.1.0/include/ignition/physics/Entity.hh:303,
                 from /home/acxz/vcs/git/github/acxz/pkgbuilds/ignition-physics/src/ign-physics-ignition-physics3_3.1.0/src/Identity.cc:18:
/home/acxz/vcs/git/github/acxz/pkgbuilds/ignition-physics/src/ign-physics-ignition-physics3_3.1.0/include/ignition/physics/detail/Entity.hh:363:5: error: no declaration matches ‘typename FeatureT::Implementation<Policy>* ignition::physics::Entity< <template-parameter-1-1>, <template-parameter-1-2> >::Interface()’
  363 |     Entity<Policy, Features>::Interface()
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/acxz/vcs/git/github/acxz/pkgbuilds/ignition-physics/src/ign-physics-ignition-physics3_3.1.0/src/Identity.cc:18:
/home/acxz/vcs/git/github/acxz/pkgbuilds/ignition-physics/src/ign-physics-ignition-physics3_3.1.0/include/ignition/physics/Entity.hh:283:58: note: candidates are: ‘template<class PolicyT, class FeaturesT> template<class F> const typename F::Implementation<ignition::physics::Entity< <template-parameter-1-1>, <template-parameter-1-2> >::Policy>* ignition::physics::Entity< <template-parameter-1-1>, <template-parameter-1-2> >::Interface() const’
  283 |       const typename F::template Implementation<Policy> *Interface() const;
      |                                                          ^~~~~~~~~
/home/acxz/vcs/git/github/acxz/pkgbuilds/ignition-physics/src/ign-physics-ignition-physics3_3.1.0/include/ignition/physics/Entity.hh:276:59: note:                 ‘template<class PolicyT, class FeaturesT> template<class FeatureT> typename FeatureT::Implementation<ignition::physics::Entity< <template-parameter-1-1>, <template-parameter-1-2> >::Policy>* ignition::physics::Entity< <template-parameter-1-1>, <template-parameter-1-2> >::Interface()’
  276 |       typename FeatureT::template Implementation<Policy> *Interface();
      |                                                           ^~~~~~~~~
/home/acxz/vcs/git/github/acxz/pkgbuilds/ignition-physics/src/ign-physics-ignition-physics3_3.1.0/include/ignition/physics/Entity.hh:216:11: note: ‘class ignition::physics::Entity< <template-parameter-1-1>, <template-parameter-1-2> >’ defined here
  216 |     class Entity
      |           ^~~~~~
In file included from /home/acxz/vcs/git/github/acxz/pkgbuilds/ignition-physics/src/ign-physics-ignition-physics3_3.1.0/include/ignition/physics/Entity.hh:303,
                 from /home/acxz/vcs/git/github/acxz/pkgbuilds/ignition-physics/src/ign-physics-ignition-physics3_3.1.0/src/Identity.cc:18:
/home/acxz/vcs/git/github/acxz/pkgbuilds/ignition-physics/src/ign-physics-ignition-physics3_3.1.0/include/ignition/physics/detail/Entity.hh:373:5: error: no declaration matches ‘const typename FeatureT::Implementation<Policy>* ignition::physics::Entity< <template-parameter-1-1>, <template-parameter-1-2> >::Interface() const’
  373 |     Entity<Policy, Features>::Interface() const
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/acxz/vcs/git/github/acxz/pkgbuilds/ignition-physics/src/ign-physics-ignition-physics3_3.1.0/src/Identity.cc:18:
/home/acxz/vcs/git/github/acxz/pkgbuilds/ignition-physics/src/ign-physics-ignition-physics3_3.1.0/include/ignition/physics/Entity.hh:283:58: note: candidates are: ‘template<class PolicyT, class FeaturesT> template<class F> const typename F::Implementation<ignition::physics::Entity< <template-parameter-1-1>, <template-parameter-1-2> >::Policy>* ignition::physics::Entity< <template-parameter-1-1>, <template-parameter-1-2> >::Interface() const’
  283 |       const typename F::template Implementation<Policy> *Interface() const;
      |                                                          ^~~~~~~~~
/home/acxz/vcs/git/github/acxz/pkgbuilds/ignition-physics/src/ign-physics-ignition-physics3_3.1.0/include/ignition/physics/Entity.hh:276:59: note:                 ‘template<class PolicyT, class FeaturesT> template<class FeatureT> typename FeatureT::Implementation<ignition::physics::Entity< <template-parameter-1-1>, <template-parameter-1-2> >::Policy>* ignition::physics::Entity< <template-parameter-1-1>, <template-parameter-1-2> >::Interface()’
  276 |       typename FeatureT::template Implementation<Policy> *Interface();
      |                                                           ^~~~~~~~~
/home/acxz/vcs/git/github/acxz/pkgbuilds/ignition-physics/src/ign-physics-ignition-physics3_3.1.0/include/ignition/physics/Entity.hh:216:11: note: ‘class ignition::physics::Entity< <template-parameter-1-1>, <template-parameter-1-2> >’ defined here
  216 |     class Entity
      |           ^~~~~~
make[2]: *** [src/CMakeFiles/ignition-physics3.dir/build.make:147: src/CMakeFiles/ignition-physics3.dir/Identity.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:388: src/CMakeFiles/ignition-physics3.dir/all] Error 2
make: *** [Makefile:160: all] Error 2
```

OS: Arch Linux
gcc: 10.2.0
ign-physics: 3.1.0 (also reproducible on latest commit of `ign-physics3` branch)

--- a/include/ignition/physics/Entity.hh
+++ b/include/ignition/physics/Entity.hh
@@ -273,14 +273,15 @@ namespace ignition
       /// \return A pointer to the implementation of FeatureT, or a nullptr if
       /// FeatureT is not available.
       protected: template <typename FeatureT>
-      typename FeatureT::template Implementation<Policy> *Interface();
+      typename FeatureT::template Implementation<PolicyT> *Interface();
 
       /// \brief Same as Interface(), but const-qualified so that const entities
       /// can request const-qualified interfaces from the implementation.
       /// \return A pointer to a const-qualified implementation of F, or a
       /// nullptr if F is not available.
-      protected: template <typename F>
-      const typename F::template Implementation<Policy> *Interface() const;
+      protected: template <typename FeatureT>
+      const typename FeatureT::template Implementation<PolicyT> *Interface()
+          const;
 
       /// \brief This is a pointer to the physics engine implementation, and it
       /// can be used by the object features to find the interfaces that they

